From: Andrew Morton <akpm@linux-foundation.org>
Subject: kbuild-fix-wimplicit-function-declaration-in-license_is_gpl_compatible-fix
Date: Wed Nov 16 03:46:25 PM PST 2022

remember "linux/"

Cc: Sam James <sam@gentoo.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/license.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/include/linux/license.h~kbuild-fix-wimplicit-function-declaration-in-license_is_gpl_compatible-fix
+++ a/include/linux/license.h
@@ -2,7 +2,7 @@
 #ifndef __LICENSE_H
 #define __LICENSE_H
 
-#include <string.h>
+#include <linux/string.h>
 
 static inline int license_is_gpl_compatible(const char *license)
 {
_
