From: Andrew Morton <akpm@linux-foundation.org>
Subject: lib-stackdepot-move-documentation-comments-to-stackdepoth-fix
Date: Mon Feb 13 10:04:04 AM PST 2023

grammar fix, per Alexander

Cc: Alexander Potapenko <glider@google.com>
Cc: Andrey Konovalov <andreyknvl@google.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/include/linux/stackdepot.h~lib-stackdepot-move-documentation-comments-to-stackdepoth-fix
+++ a/include/linux/stackdepot.h
@@ -11,7 +11,7 @@
  * SLUB_DEBUG needs 256 bytes per object for that). Since allocation and free
  * stack traces often repeat, using stack depot allows to save about 100x space.
  *
- * Stack traces are never removed from stack depot.
+ * Stack traces are never removed from the stack depot.
  *
  * Author: Alexander Potapenko <glider@google.com>
  * Copyright (C) 2016 Google, Inc.
_
