From: Andrew Morton <akpm@linux-foundation.org>
Subject: lib-stackdepot-mm-rename-stack_depot_want_early_init-fix
Date: Fri Feb 10 03:13:49 PM PST 2023

update mm/kmemleak.c

Cc: Alexander Potapenko <glider@google.com>
Cc: Andrey Konovalov <andreyknvl@google.com>
Cc: Vlastimil Babka <vbabka@suse.cz>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/mm/kmemleak.c~lib-stackdepot-mm-rename-stack_depot_want_early_init-fix
+++ a/mm/kmemleak.c
@@ -2071,7 +2071,7 @@ static int __init kmemleak_boot_config(c
 		kmemleak_disable();
 	else if (strcmp(str, "on") == 0) {
 		kmemleak_skip_disable = 1;
-		stack_depot_want_early_init();
+		stack_depot_request_early_init();
 	}
 	else
 		return -EINVAL;
_
