From: Andrew Morton <akpm@linux-foundation.org>
Subject: MAINTAINERS: add lib/raid6/ to "SOFTWARE RAID"
Date: Thu Jun 19 04:13:17 PM PDT 2025

Cc: Song Liu <song@kernel.org>
Cc: Yu Kuai <yukuai3@huawei.com>
Cc: Herbert Xu <herbert@gondor.apana.org.au>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 MAINTAINERS |    1 +
 1 file changed, 1 insertion(+)

--- a/MAINTAINERS~maintainers-add-lib-raid6-to-software-raid
+++ a/MAINTAINERS
@@ -23052,6 +23052,7 @@ F:	drivers/md/md*
 F:	drivers/md/raid*
 F:	include/linux/raid/
 F:	include/uapi/linux/raid/
+F:	lib/raid6/
 
 SOLIDRUN CLEARFOG SUPPORT
 M:	Russell King <linux@armlinux.org.uk>
_
