From: Andrew Morton <akpm@linux-foundation.org>
Subject: maintainers-add-page-cache-reviewer-fix
Date: Wed Apr 15 10:47:13 PM PDT 2026

add linux-mm@kvack.org

Cc: Jan Kara <jack@suse.cz>
Cc: Matthew Wilcox (Oracle) <willy@infradead.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 MAINTAINERS |    1 +
 1 file changed, 1 insertion(+)

--- a/MAINTAINERS~maintainers-add-page-cache-reviewer-fix
+++ a/MAINTAINERS
@@ -19966,6 +19966,7 @@ PAGE CACHE
 M:	Matthew Wilcox (Oracle) <willy@infradead.org>
 R:	Jan Kara <jack@suse.cz>
 L:	linux-fsdevel@vger.kernel.org
+L:	linux-mm@kvack.org
 S:	Supported
 T:	git git://git.infradead.org/users/willy/pagecache.git
 F:	Documentation/filesystems/locking.rst
_
