From: Andrew Morton <akpm@linux-foundation.org>
Subject: maintainers-remove-hugetlb-maintainer-mike-kravetz-fix
Date: Wed Dec 20 02:23:04 PM PST 2023

update CREDITS

Cc: Mike Kravetz <mike.kravetz@oracle.com>
Cc: Muchun Song <songmuchun@bytedance.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 CREDITS |    4 ++++
 1 file changed, 4 insertions(+)

--- a/CREDITS~maintainers-remove-hugetlb-maintainer-mike-kravetz-fix
+++ a/CREDITS
@@ -2126,6 +2126,10 @@ S: 2213 La Terrace Circle
 S: San Jose, CA 95123
 S: USA
 
+N: Mike Kravetz
+E: mike.kravetz@oracle.com
+D: Maintenance and development of the hugetlb subsystem
+
 N: Andreas S. Krebs
 E: akrebs@altavista.net
 D: CYPRESS CY82C693 chipset IDE, Digital's PC-Alpha 164SX boards
_
