From: Matthew Wilcox <willy@infradead.org>
Subject: mm-add-folio_add_new_anon_rmap-fix-2
Date: Wed, 11 Jan 2023 21:44:33 +0000

folio-compat.c needs rmap.h

Signed-off-by: Matthew Wilcox <willy@infradead.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/mm/folio-compat.c~mm-add-folio_add_new_anon_rmap-fix-2
+++ a/mm/folio-compat.c
@@ -6,6 +6,7 @@
 
 #include <linux/migrate.h>
 #include <linux/pagemap.h>
+#include <linux/rmap.h>
 #include <linux/swap.h>
 #include "internal.h"
 
_
