From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-add-memcpy_from_file_folio-fix
Date: Thu Jan 26 03:36:39 PM PST 2023

Cc: "Fabio M. De Francesco" <fmdefrancesco@gmail.com>
Cc: Ira Weiny <ira.weiny@intel.com>
Cc: "Matthew Wilcox (Oracle)" <willy@infradead.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/highmem.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/include/linux/highmem.h~mm-add-memcpy_from_file_folio-fix
+++ a/include/linux/highmem.h
@@ -432,7 +432,7 @@ static inline size_t memcpy_from_file_fo
 	char *from = kmap_local_folio(folio, offset);
 
 	if (folio_test_highmem(folio))
-		len = min(len, PAGE_SIZE - offset);
+		len = min_t(size_t, len, PAGE_SIZE - offset);
 	else
 		len = min(len, folio_size(folio) - offset);
 
_
