From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-fix-for-negative-counter-nr_file_hugepages-fix
Date: Mon Nov  6 11:44:47 AM PST 2023

whitespace fix

Cc: Johannes Weiner <hannes@cmpxchg.org>
Cc: Rik van Riel <riel@surriel.com>
Cc: Stefan Roesch <shr@devkernel.io>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/huge_memory.c |    1 -
 1 file changed, 1 deletion(-)

--- a/mm/huge_memory.c~mm-fix-for-negative-counter-nr_file_hugepages-fix
+++ a/mm/huge_memory.c
@@ -2773,7 +2773,6 @@ int split_huge_page_to_list(struct page
 				__lruvec_stat_mod_folio(folio, NR_SHMEM_THPS,
 							-nr);
 			} else if (folio_test_pmd_mappable(folio)) {
-
 				__lruvec_stat_mod_folio(folio, NR_FILE_THPS,
 							-nr);
 				filemap_nr_thps_dec(mapping);
_
