From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-memremap-introduce-pgmap_request_folio-using-pgmap-offsets-fix-2
Date: Thu Oct 27 11:53:07 AM PDT 2022

fix whitespace, per Felix and Yang Li

Cc: Felix Kuehling <Felix.Kuehling@amd.com>
Cc: Yang Li <yang.lee@linux.alibaba.com>
Cc: Dan Williams <dan.j.williams@intel.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/drivers/gpu/drm/amd/amdkfd/kfd_migrate.c~mm-memremap-introduce-pgmap_request_folio-using-pgmap-offsets-fix-2
+++ a/drivers/gpu/drm/amd/amdkfd/kfd_migrate.c
@@ -328,8 +328,8 @@ svm_migrate_copy_to_vram(struct amdgpu_d
 
 		dst[i] = cursor.start + (j << PAGE_SHIFT);
 		migrate->dst[i] = svm_migrate_addr_to_pfn(adev, dst[i]);
-			svm_migrate_get_vram_page(&kfddev->pgmap, prange,
-						  migrate->dst[i]);
+		svm_migrate_get_vram_page(&kfddev->pgmap, prange,
+					  migrate->dst[i]);
 		migrate->dst[i] = migrate_pfn(migrate->dst[i]);
 
 		spage = migrate_pfn_to_page(migrate->src[i]);
_
