From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm/page_alloc.c: document bulkfree_pcp_prepare() return value
Date: Tue Sep 13 03:30:38 PM PDT 2022

Cc: Catalin Marinas <catalin.marinas@arm.com>
Cc: ke.wang <ke.wang@unisoc.com>
Cc: Matthew Wilcox <willy@infradead.org>
Cc: Zhaoyang Huang <huangzhaoyang@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/page_alloc.c |    1 +
 1 file changed, 1 insertion(+)

--- a/mm/page_alloc.c~mm-page_allocc-document-bulkfree_pcp_prepare-return-value
+++ a/mm/page_alloc.c
@@ -1507,6 +1507,7 @@ static bool free_pcp_prepare(struct page
 	return free_pages_prepare(page, order, true, FPI_NONE);
 }
 
+/* return true if this page has an inappropriate state */
 static bool bulkfree_pcp_prepare(struct page *page)
 {
 	if (debug_pagealloc_enabled_static())
_
