From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-rmap-mlock-large-folios-in-try_to_unmap_one-fix
Date: Tue Sep 23 02:28:52 PM PDT 2025

s/CROSSSED/CROSSED/

Cc: Baolin Wang <baolin.wang@linux.alibaba.com>
Cc: David Hildenbrand <david@redhat.com>
Cc: Johannes Weiner <hannes@cmpxchg.org>
Cc: Kiryl Shutsemau <kas@kernel.org>
Cc: Lorenzo Stoakes <lorenzo.stoakes@oracle.com>
Cc: Shakeel Butt <shakeel.butt@linux.dev>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/rmap.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/rmap.c~mm-rmap-mlock-large-folios-in-try_to_unmap_one-fix
+++ a/mm/rmap.c
@@ -1933,7 +1933,7 @@ static bool try_to_unmap_one(struct foli
 			 * If page table boundary has been cross, current ptl
 			 * only protect part of ptes.
 			 */
-			if (pvmw.flags & PVMW_PGTABLE_CROSSSED)
+			if (pvmw.flags & PVMW_PGTABLE_CROSSED)
 				goto walk_done;
 
 			/* Restore the mlock which got missed */
_
