From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-replace-vma-vm_flags-direct-modifications-with-modifier-calls-fix
Date: Wed Jan 25 05:47:16 PM PST 2023

fix mm/nommu.c

Cc: Michal Hocko <mhocko@suse.com>
Cc: Suren Baghdasaryan <surenb@google.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/mm/nommu.c~mm-replace-vma-vm_flags-direct-modifications-with-modifier-calls-fix
+++ a/mm/nommu.c
@@ -951,7 +951,7 @@ static int do_mmap_private(struct vm_are
 	atomic_long_add(total, &mmap_pages_allocated);
 
 	set_vm_flags(vma, VM_MAPPED_COPY);
-	region->vm_flags = vma->flags;
+	region->vm_flags = vma->vm_flags;
 	region->vm_start = (unsigned long) base;
 	region->vm_end   = region->vm_start + len;
 	region->vm_top   = region->vm_start + (total << PAGE_SHIFT);
_
