From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-shrinkers-fix-deadlock-in-shrinker-debugfs-fix
Date: Thu Feb  2 04:31:34 PM PST 2023

fix CONFIG_SHRINKER_DEBUG=n stub

Cc: Kent Overstreet <kent.overstreet@gmail.com>
Cc: Muchun Song <songmuchun@bytedance.com>
Cc: Qi Zheng <zhengqi.arch@bytedance.com>
Cc: Roman Gushchin <roman.gushchin@linux.dev>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/include/linux/shrinker.h~mm-shrinkers-fix-deadlock-in-shrinker-debugfs-fix
+++ a/include/linux/shrinker.h
@@ -117,6 +117,7 @@ static inline int shrinker_debugfs_add(s
 }
 static inline struct dentry *shrinker_debugfs_remove(struct shrinker *shrinker)
 {
+	return NULL;
 }
 static inline __printf(2, 3)
 int shrinker_debugfs_rename(struct shrinker *shrinker, const char *fmt, ...)
_
