From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-shrinkers-make-count-and-scan-in-shrinker-debugfs-lockless-fix
Date: Thu Mar  9 12:52:15 PM PST 2023

include linux/srcu.h, per Vlaistimil

Cc: Qi Zheng <zhengqi.arch@bytedance.com>
Cc: Vlastimil Babka <vbabka@suse.cz>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/mm/shrinker_debug.c~mm-shrinkers-make-count-and-scan-in-shrinker-debugfs-lockless-fix
+++ a/mm/shrinker_debug.c
@@ -2,6 +2,7 @@
 #include <linux/idr.h>
 #include <linux/slab.h>
 #include <linux/debugfs.h>
+#include <linux/srcu.h>
 #include <linux/seq_file.h>
 #include <linux/shrinker.h>
 #include <linux/memcontrol.h>
_
