From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-vmscan-extract-calculated-pressure-balance-as-a-function-fix
Date: Tue Jan 28 08:13:57 PM PST 2025

80-col wrapping

Cc: Hao Zhang <zhanghao1@kylinos.cn>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/vmscan.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/mm/vmscan.c~mm-vmscan-extract-calculated-pressure-balance-as-a-function-fix
+++ a/mm/vmscan.c
@@ -2400,8 +2400,8 @@ static void prepare_scan_control(pg_data
 	}
 }
 
-static inline void calculate_pressure_balance(struct scan_control *sc, int swappiness,
-			u64 *fraction, u64 *denominator)
+static inline void calculate_pressure_balance(struct scan_control *sc,
+			int swappiness, u64 *fraction, u64 *denominator)
 {
 	unsigned long anon_cost, file_cost, total_cost;
 	unsigned long ap, fp;
_
