From: Andrew Morton <akpm@linux-foundation.org>
Subject: ocfs2-build-hack
Date: Sun Nov  9 11:29:24 AM PST 2025

Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 fs/ocfs2/xattr.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/fs/ocfs2/xattr.c~ocfs2-build-hack
+++ a/fs/ocfs2/xattr.c
@@ -6362,8 +6362,8 @@ static int ocfs2_reflink_xattr_header(ha
 			if (new_xh->xh_count) {
 				memmove(xe, xe + 1,
 					(void *)last - (void *)xe);
-				memset(last, 0,
-				       sizeof(struct ocfs2_xattr_entry));
+//				memset(last, 0,
+//				       sizeof(struct ocfs2_xattr_entry));
 			}
 
 			/*
_
