From: Andrew Morton <akpm@linux-foundation.org>
Subject: resource-fix-resource-leak-in-get_free_mem_region-fix
Date: Tue Mar  4 04:11:59 PM PST 2025

fix braceology

Cc: Andriy Shevchenko <andriy.shevchenko@linux.intel.com>
Cc: Bjorn Helgaas <bhelgaas@google.com>
Cc: Dan Wiliams <dan.j.williams@intel.com>
Cc: "Huang, Ying" <huang.ying.caritas@gmail.com>
Cc: Ilpo Jarvinen <ilpo.jarvinen@linux.intel.com>
Cc: Joanthan Cameron <Jonathan.Cameron@huawei.com>
Cc: Li Zhijian <lizhijian@fujitsu.com>
Cc: Mika Westeberg <mika.westerberg@linux.intel.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 kernel/resource.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/kernel/resource.c~resource-fix-resource-leak-in-get_free_mem_region-fix
+++ a/kernel/resource.c
@@ -1998,10 +1998,11 @@ get_free_mem_region(struct device *dev,
 	if (flags & GFR_REQUEST_REGION) {
 		free_resource(res);
 		devres_free(dr);
-	} else if (dev)
+	} else if (dev) {
 		devm_release_action(dev, remove_free_mem_region, res);
-	else
+	} else {
 		free_resource(res);
+	}
 
 	return ERR_PTR(-ERANGE);
 }
_
