From: Jesper Juhl <jesperjuhl76@gmail.com>
Subject: Update CREDITS file entry for Jesper Juhl

Link: https://lkml.kernel.org/r/8e18fa82-1095-330f-29fb-80743d22443c@gmail.com
Signed-off-by: Jesper Juhl <jesperjuhl76@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/CREDITS~update-credits-file-entry-for-jesper-juhl
+++ a/CREDITS
@@ -1848,11 +1848,11 @@ E: ajoshi@shell.unixbox.com
 D: fbdev hacking
 
 N: Jesper Juhl
-E: jj@chaosbits.net
+E: jesperjuhl76@gmail.com
 D: Various fixes, cleanups and minor features all over the tree.
 D: Wrote initial version of the hdaps driver (since passed on to others).
-S: Lemnosvej 1, 3.tv
-S: 2300 Copenhagen S.
+S: Titangade 5G, 2.tv
+S: 2200 Copenhagen N.
 S: Denmark
 
 N: Jozsef Kadlecsik
_
