From: Andrew Morton <akpm@linux-foundation.org>
Subject: vma-detect-infinite-loop-in-vma-tree-fix
Date: Thu Oct 31 01:45:34 PM PDT 2024

add comment

Cc: Jann Horn <jannh@google.com>
Cc: "Liam R. Howlett" <Liam.Howlett@oracle.com>
Cc: Lorenzo Stoakes <lorenzo.stoakes@oracle.com>
Cc: Vlastimil Babka <vbabka@suse.cz>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/vma.c |    1 +
 1 file changed, 1 insertion(+)

--- a/mm/vma.c~vma-detect-infinite-loop-in-vma-tree-fix
+++ a/mm/vma.c
@@ -615,6 +615,7 @@ void validate_mm(struct mm_struct *mm)
 			anon_vma_unlock_read(anon_vma);
 		}
 #endif
+		/* Check for a infinite loop */
 		if (++i > mm->map_count + 10) {
 			i = -1;
 			break;
_
