From: Andrew Morton <akpm@linux-foundation.org>
Subject: vmcoreinfo-make-hwerr_data-visible-for-debugging-fix
Date: Wed Jan 21 12:23:52 PM PST 2026

add comment

Cc: Baoquan He <bhe@redhat.com>
Cc: Breno Leitao <leitao@debian.org>
Cc: Dave Young <dyoung@redhat.com>
Cc: "Luck, Tony" <tony.luck@intel.com>
Cc: Omar Sandoval <osandov@osandov.com>
Cc: Shuai Xue <xueshuai@linux.alibaba.com>
Cc: Vivek Goyal <vgoyal@redhat.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 kernel/vmcore_info.c |    1 +
 1 file changed, 1 insertion(+)

--- a/kernel/vmcore_info.c~vmcoreinfo-make-hwerr_data-visible-for-debugging-fix
+++ a/kernel/vmcore_info.c
@@ -36,6 +36,7 @@ struct hwerr_info {
 	time64_t timestamp;
 };
 
+/* hwerr_data[] has global scope to make it available in vmcoreinfo under LTO */
 struct hwerr_info hwerr_data[HWERR_RECOV_MAX];
 
 Elf_Word *append_elf_note(Elf_Word *buf, char *name, unsigned int type,
_
