From: Andrew Morton <akpm@linux-foundation.org>
Subject: x86-uaccess-avoid-check_object_size-in-copy_from_user_nmi-fix
Date: Wed Sep 21 05:50:47 PM PDT 2022

no instrument_copy_from_user() in my tree...

Cc: Kees Cook <keescook@chromium.org>
Cc: Yu Zhao <yuzhao@google.com>
Cc: Florian Lehner <dev@der-flo.net>
Cc: Andrew Morton <akpm@linux-foundation.org>
Cc: Peter Zijlstra (Intel) <peterz@infradead.org>
Cc: Florian Lehner <dev@der-flo.net>
Cc: Matthew Wilcox <willy@infradead.org>
Cc: Josh Poimboeuf <jpoimboe@kernel.org>
Cc: Dave Hansen <dave.hansen@linux.intel.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/arch/x86/lib/usercopy.c~x86-uaccess-avoid-check_object_size-in-copy_from_user_nmi-fix
+++ a/arch/x86/lib/usercopy.c
@@ -44,7 +44,6 @@ copy_from_user_nmi(void *to, const void
 	 * called from other contexts.
 	 */
 	pagefault_disable();
-	instrument_copy_from_user(to, from, n);
 	ret = raw_copy_from_user(to, from, n);
 	pagefault_enable();
 
_
