From: Andrew Morton <akpm@linux-foundation.org>
Subject: x86-kexec-add-a-sanity-check-on-previous-kernels-ima-kexec-buffer-fix
Date: Mon Dec  1 09:20:28 AM PST 2025

fix comment typo

Cc: Alexander Graf <graf@amazon.com>
Cc: Ard Biesheuvel <ardb@kernel.org>
Cc: Borislav Betkov <bp@alien8.de>
Cc: guoweikang <guoweikang.kernel@gmail.com>
Cc: Harshit Mogalapalli <harshit.m.mogalapalli@oracle.com>
Cc: Henry Willard <henry.willard@oracle.com>
Cc: "H. Peter Anvin" <hpa@zytor.com>
Cc: Ingo Molnar <mingo@redhat.com>
Cc: Jiri Bohac <jbohac@suse.cz>
Cc: Joel Granados <joel.granados@kernel.org>
Cc: Jonathan McDowell <noodles@meta.com>
Cc: Mike Rapoport <rppt@kernel.org>
Cc: Mimi Zohar <zohar@linux.ibm.com>
Cc: Paul Webb <paul.x.webb@oracle.com>
Cc: Sohil Mehta <sohil.mehta@intel.com>
Cc: Sourabh Jain <sourabhjain@linux.ibm.com>
Cc: Thomas Gleinxer <tglx@linutronix.de>
Cc: Yifei Liu <yifei.l.liu@oracle.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/arch/x86/kernel/setup.c~x86-kexec-add-a-sanity-check-on-previous-kernels-ima-kexec-buffer-fix
+++ a/arch/x86/kernel/setup.c
@@ -446,7 +446,7 @@ int __init ima_get_kexec_buffer(void **a
 
 	/*
 	 * Calculate the PFNs for the buffer and ensure
-	 * they are with in addressable memory.
+	 * they are within addressable memory.
 	 */
 	start_pfn = PFN_DOWN(ima_kexec_buffer_phys);
 	end_pfn = PFN_DOWN(ima_kexec_buffer_phys + ima_kexec_buffer_size - 1);
_
