From: Andrew Morton <akpm@linux-foundation.org>
Subject: zsmalloc-zs_object_copy-add-clarifying-comment-fix
Date: Thu Aug 11 02:20:45 PM PDT 2022

tweak comment layout

Cc: Alexey Romanov <avromanov@sberdevices.ru>
Cc: Minchan Kim <minchan@kernel.org>
Cc: Nitin Gupta <ngupta@vflare.org>
Cc: Sergey Senozhatsky <senozhatsky@chromium.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/zsmalloc.c |    9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

--- a/mm/zsmalloc.c~zsmalloc-zs_object_copy-add-clarifying-comment-fix
+++ a/mm/zsmalloc.c
@@ -1555,10 +1555,11 @@ static void zs_object_copy(struct size_c
 		d_off += size;
 		d_size -= size;
 
-		/* Calling kunmap_atomic(d_addr) is necessary. kunmap_atomic() calls
-		 * must occurs in reverse order of calls to kmap_atomic().
-		 * So, to call kunmap_atomic(s_addr) we should first call kunmap_atomic(d_addr).
-		 * For more details see:
+		/*
+		 * Calling kunmap_atomic(d_addr) is necessary. kunmap_atomic()
+		 * calls must occurs in reverse order of calls to kmap_atomic().
+		 * So, to call kunmap_atomic(s_addr) we should first call
+		 * kunmap_atomic(d_addr).  For more details see
 		 * https://lore.kernel.org/linux-mm/5512421D.4000603@samsung.com/
 		 */
 		if (s_off >= PAGE_SIZE) {
_
