From: Andrew Morton <akpm@linux-foundation.org>
Subject: fsdax-dax_unshare_iter-should-return-a-valid-length-fix
Date: Thu Feb  2 04:15:00 PM PST 2023

don't mess up `ret', per Matthew

Cc: Alistair Popple <apopple@nvidia.com>
Cc: Dan Williams <dan.j.williams@intel.com>
Cc: Darrick J. Wong <djwong@kernel.org>
Cc: Dave Chinner <david@fromorbit.com>
Cc: Jason Gunthorpe <jgg@nvidia.com>
Cc: John Hubbard <jhubbard@nvidia.com>
Cc: Shiyang Ruan <ruansy.fnst@fujitsu.com>
Cc: Matthew Wilcox <willy@infradead.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/fs/dax.c~fsdax-dax_unshare_iter-should-return-a-valid-length-fix
+++ a/fs/dax.c
@@ -1271,10 +1271,10 @@ static s64 dax_unshare_iter(struct iomap
 	if (ret < 0)
 		goto out_unlock;
 
-	ret = copy_mc_to_kernel(daddr, saddr, length);
-	if (ret)
+	if (copy_mc_to_kernel(daddr, saddr, length) == 0)
+		ret = length;
+	else
 		ret = -EIO;
-	ret = length;
 
 out_unlock:
 	dax_read_unlock(id);
_
