From: Andrew Morton <akpm@linux-foundation.org>
Subject: lib-alloc_tag_module_unload-must-wait-for-pending-kfree_rcu-calls-fix
Date: Mon Oct  7 06:06:19 PM PDT 2024

add comment

Cc: Ben Greear <greearb@candelatech.com>
Cc: Florian Westphal <fw@strlen.de>
Cc: Kent Overstreet <kent.overstreet@linux.dev>
Cc: Suren Baghdasaryan <surenb@google.com>
Cc: Uladzislau Rezki <urezki@gmail.com>
Cc: Vlastimil Babka <vbabka@suse.cz>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/codetag.c |    1 +
 1 file changed, 1 insertion(+)

--- a/lib/codetag.c~lib-alloc_tag_module_unload-must-wait-for-pending-kfree_rcu-calls-fix
+++ a/lib/codetag.c
@@ -228,6 +228,7 @@ bool codetag_unload_module(struct module
 	if (!mod)
 		return true;
 
+	/* await any module's kfree_rcu() operations to complete */
 	kvfree_rcu_barrier();
 
 	mutex_lock(&codetag_lock);
_
