From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-add-max-swappiness-arg-to-lru_gen-for-anonymous-memory-only-update-fix
Date: Wed May  7 06:27:25 PM PDT 2025

tweak coding style

Cc: Zhongkun He <hezhongkun.hzk@bytedance.com>
Cc: Johannes Weiner <hannes@cmpxchg.org>
Cc: Michal Hocko <mhocko@suse.com>
Cc: Muchun Song <muchun.song@linux.dev>
Cc: Yosry Ahmed <yosry.ahmed@linux.dev>
Cc: Yu Zhao <yuzhao@google.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/vmscan.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/mm/vmscan.c~mm-add-max-swappiness-arg-to-lru_gen-for-anonymous-memory-only-update-fix
+++ a/mm/vmscan.c
@@ -5606,9 +5606,9 @@ static ssize_t lru_gen_seq_write(struct
 			break;
 		}
 
-		if (n == 4)
+		if (n == 4) {
 			swappiness = -1;
-		else if (!strcmp("max", swap_string)) {
+		} else if (!strcmp("max", swap_string)) {
 			/* set by userspace for anonymous memory only */
 			swappiness = SWAPPINESS_ANON_ONLY;
 		} else {
_
