From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-add-remap_pfn_range_prepare-remap_pfn_range_complete-fix
Date: Thu Sep 18 02:38:36 PM PDT 2025

restore inadvertently-removed newline

Cc: Jason Gunthorpe <jgg@nvidia.com>
Cc: Lorenzo Stoakes <lorenzo.stoakes@oracle.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/memory.c |    1 +
 1 file changed, 1 insertion(+)

--- a/mm/memory.c~mm-add-remap_pfn_range_prepare-remap_pfn_range_complete-fix
+++ a/mm/memory.c
@@ -2955,6 +2955,7 @@ static int remap_pfn_range_notrack(struc
 		unsigned long pfn, unsigned long size, pgprot_t prot)
 {
 	int error = remap_pfn_range_internal(vma, addr, pfn, size, prot);
+
 	if (!error)
 		return 0;
 
_
