From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-change-to-return-bool-for-isolate_movable_page-fix
Date: Thu Feb 16 02:41:42 PM PST 2023

remove unneeded comment, per Matthew

Cc: Baolin Wang <baolin.wang@linux.alibaba.com>
Cc: David Hildenbrand <david@redhat.com>
Cc: Linus Torvalds <torvalds@linux-foundation.org>
Cc: Matthew Wilcox (Oracle) <willy@infradead.org>
Cc: SeongJae Park <sj@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/mm/memory_hotplug.c~mm-change-to-return-bool-for-isolate_movable_page-fix
+++ a/mm/memory_hotplug.c
@@ -1672,7 +1672,7 @@ do_migrate_range(unsigned long start_pfn
 			isolated = isolate_lru_page(page);
 		else
 			isolated = isolate_movable_page(page, ISOLATE_UNEVICTABLE);
-		if (isolated) { /* Success */
+		if (isolated) {
 			list_add_tail(&page->lru, &source);
 			if (!__PageMovable(page))
 				inc_node_page_state(page, NR_ISOLATED_ANON +
_
