From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-hugetlb-add-hugetlb_alloc_threads-cmdline-option-fix
Date: Mon Mar  3 08:51:01 PM PST 2025

tidy up a comment

Cc: Jonathan Corbet <corbet@lwn.net>
Cc: Muchun Song <muchun.song@linux.dev>
Cc: Thomas Prescher <thomas.prescher@cyberus-technology.de>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/hugetlb.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/mm/hugetlb.c~mm-hugetlb-add-hugetlb_alloc_threads-cmdline-option-fix
+++ a/mm/hugetlb.c
@@ -5002,7 +5002,9 @@ void __init hugetlb_bootmem_alloc(void)
 	__hugetlb_bootmem_allocated = true;
 }
 
-/* hugepage_alloc_threads command line parsing
+/*
+ * hugepage_alloc_threads command line parsing.
+ *
  * When set, use this specific number of threads for the boot
  * allocation of hugepages.
  */
_
