From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-khugepaged-introduce-khugepaged_enter_vma-helper-vs-maple-tree

Cc: Liam Howlett <liam.howlett@oracle.com>
Cc: Yang Shi <shy828301@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/mmap.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/mm/mmap.c~mm-khugepaged-introduce-khugepaged_enter_vma-helper-vs-maple-tree
+++ a/mm/mmap.c
@@ -2580,7 +2580,7 @@ unsigned long mmap_region(struct file *f
 	/* Actually expand, if possible */
 	if (vma &&
 	    !vma_expand(&mas, vma, merge_start, merge_end, vm_pgoff, next)) {
-		khugepaged_enter_vma_merge(vma, vm_flags);
+		khugepaged_enter_vma(vma, vm_flags);
 		goto expanded;
 	}
 
@@ -2995,7 +2995,7 @@ static int do_brk_flags(struct ma_state
 			anon_vma_interval_tree_post_update_vma(vma);
 			anon_vma_unlock_write(vma->anon_vma);
 		}
-		khugepaged_enter_vma_merge(vma, flags);
+		khugepaged_enter_vma(vma, flags);
 		goto out;
 	}
 
_
