From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-remove-the-vma-linked-list-fix-fix-fix
Date: Fri Jun  3 08:44:15 AM PDT 2022

fix merge issue

Cc: Liam Howlett <liam.howlett@oracle.com>
Cc: Suren Baghdasaryan <surenb@google.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/mmap.c |    1 -
 1 file changed, 1 deletion(-)

--- a/mm/mmap.c~mm-remove-the-vma-linked-list-fix-fix-fix
+++ a/mm/mmap.c
@@ -3274,7 +3274,6 @@ void exit_mmap(struct mm_struct *mm)
 
 	trace_exit_mmap(mm);
 	__mt_destroy(&mm->mm_mt);
-	mm->mmap = NULL;
 	mmap_write_unlock(mm);
 	vm_unacct_memory(nr_accounted);
 }
_
