From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-replace-vm_warn_on-to-pr_warn-if-the-node-is-offline-with-__gfp_thisnode-v4-fix
Date: Wed Nov 23 11:56:07 AM PST 2022

fix whitespace

Cc: Yang Shi <shy828301@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/gfp.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/include/linux/gfp.h~mm-replace-vm_warn_on-to-pr_warn-if-the-node-is-offline-with-__gfp_thisnode-v4-fix
+++ a/include/linux/gfp.h
@@ -217,7 +217,7 @@ static inline void warn_if_node_offline(
 	if (warn_gfp != (__GFP_THISNODE|__GFP_NOWARN))
 		return;
 
-	if(node_online(this_node))
+	if (node_online(this_node))
 		return;
 
 	pr_warn("%pGg allocation from offline node %d\n", &warn_gfp, this_node);
_
