From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-shmem-fix-khugepaged-activation-policy-for-shmem-fix-fix
Date: Fri Sep 20 03:19:14 AM PDT 2024

fix the fix

Cc: Baolin Wang <baolin.wang@linux.alibaba.com>
Cc: David Hildenbrand <david@redhat.com>
Cc: Hugh Dickins <hughd@google.com>
Cc: kernel test robot <lkp@intel.com>
Cc: Matthew Wilcox (Oracle) <willy@infradead.org>
Cc: Ryan Roberts <ryan.roberts@arm.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/shmem.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/shmem.c~mm-shmem-fix-khugepaged-activation-policy-for-shmem-fix-fix
+++ a/mm/shmem.c
@@ -1652,7 +1652,7 @@ static gfp_t limit_gfp_mask(gfp_t huge_g
 	return result;
 }
 
-#ifdef CONFIG_TRANSPARENT_HUGEPAGE
+#if defined(CONFIG_TRANSPARENT_HUGEPAGE) && defined(SHMEM)
 bool shmem_hpage_pmd_enabled(void)
 {
 	if (shmem_huge == SHMEM_HUGE_DENY)
_
