From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-shmem-add-support-for-fs_ioc_etflags-for-tmpfs-fix
Date: Wed Jul 13 08:48:24 AM PDT 2022

remove extra newlines

Cc: "Theodore Ts'o" <tytso@mit.edu>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/shmem.c |    2 --
 1 file changed, 2 deletions(-)

--- a/mm/shmem.c~mm-shmem-add-support-for-fs_ioc_etflags-for-tmpfs-fix
+++ a/mm/shmem.c
@@ -2846,8 +2846,6 @@ int shmem_fileattr_get(struct dentry *de
 	return 0;
 }
 
-
-
 int shmem_fileattr_set(struct user_namespace *mnt_userns,
 		       struct dentry *dentry, struct fileattr *fa)
 {
_
