From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-vmalloc-convert-vread-to-vread_iter-fix
Date: Tue Mar 21 02:34:51 PM PDT 2023

fix nommu build

Cc: Lorenzo Stoakes <lstoakes@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/mm/nommu.c~mm-vmalloc-convert-vread-to-vread_iter-fix
+++ a/mm/nommu.c
@@ -199,7 +199,7 @@ unsigned long vmalloc_to_pfn(const void
 }
 EXPORT_SYMBOL(vmalloc_to_pfn);
 
-long vread_iter(struct iov_iter *iter, char *addr, size_t count)
+long vread_iter(struct iov_iter *iter, const char *addr, size_t count)
 {
 	/* Don't allow overflow */
 	if ((unsigned long) addr + count < count)
_
