From: Andrew Morton <akpm@linux-foundation.org>
Subject: proc-rewrite-next_tgid-fix
Date: Thu Apr 23 06:52:34 AM PDT 2026

remove newline

Cc: Alexey Dobriyan <adobriyan@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 fs/proc/base.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/fs/proc/base.c~proc-rewrite-next_tgid-fix
+++ a/fs/proc/base.c
@@ -3546,8 +3546,7 @@ struct tgid_iter {
 	struct pid_namespace *const pid_ns;
 };
 
-static
-struct tgid_iter
+static struct tgid_iter
 make_tgid_iter(unsigned int init_tgid, struct pid_namespace *pid_ns)
 {
 	return (struct tgid_iter){
_
