From: Andrew Morton <akpm@linux-foundation.org>
Subject: shmem-userfaultfd-implement-shmem-uffd-operations-using-vm_uffd_ops-fix-fix
Date: Tue Mar 31 09:04:03 PM PDT 2026

include file.h for fput()

Cc: David Carlier <devnexen@gmail.com>
Cc: Mike Rapoport <rppt@kernel.org>
Cc: Peter Xu <peterx@redhat.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/userfaultfd.c |    1 +
 1 file changed, 1 insertion(+)

--- a/mm/userfaultfd.c~shmem-userfaultfd-implement-shmem-uffd-operations-using-vm_uffd_ops-fix-fix
+++ a/mm/userfaultfd.c
@@ -11,6 +11,7 @@
 #include <linux/rmap.h>
 #include <linux/swap.h>
 #include <linux/leafops.h>
+#include <linux/file.h>
 #include <linux/userfaultfd_k.h>
 #include <linux/mmu_notifier.h>
 #include <linux/hugetlb.h>
_
