From: Andrew Morton <akpm@linux-foundation.org>
Subject: zsmalloc-fine-grained-inuse-ratio-based-fullness-grouping-fix
Date: Fri Mar  3 03:24:35 PM PST 2023

fix comment spello

Cc: Sergey Senozhatsky <senozhatsky@chromium.org>
Cc: Minchan Kim <minchan@kernel.org>
Cc: Yosry Ahmed <yosryahmed@google.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/mm/zsmalloc.c~zsmalloc-fine-grained-inuse-ratio-based-fullness-grouping-fix
+++ a/mm/zsmalloc.c
@@ -647,7 +647,7 @@ static int zs_stats_size_show(struct seq
 		spin_lock(&pool->lock);
 
 		/*
-		 * Replecate old behaviour for almost_full and almost_empty
+		 * Replicate old behaviour for almost_full and almost_empty
 		 * stats.
 		 */
 		for (fg = ZS_INUSE_RATIO_70; fg <= ZS_INUSE_RATIO_99; fg++)
_
